vulnerabilities/importers/pysec.py

```diff
@@ -0,0 +1,240 @@
+    )
+
+
+def fixed_filter(fixed_range) -> []:
+    """
+    Return a list of fixed version strings given a ``fixed_range`` mapping of OSV data.
+    >>> list(fixed_filter({"type": "SEMVER", "events": [{"introduced": "0"}, {"fixed": "1.6.0"}]}))
+    ['1.6.0']
+    >>> list(fixed_filter({"type": "ECOSYSTEM","events":[{"introduced": "0"},{"fixed": "1.0.0"},{"fixed": "9.0.0"}]}))
+    ['1.0.0', '9.0.0']
+    """
+    for event in fixed_range.get("events") or []:
+        fixed = event.get("fixed")
+        if fixed:
+            yield fixed
+
+
+def get_aliases(raw_data) -> []:
+    """
+    aliases field is optional , id is required and these are all aliases from our perspective
+    converting list of two fields to a dict then , convert it to a list to make sure a list is unique
+    >>> get_aliases({"id": "GHSA-j3f7-7rmc-6wqj"})
+    ['GHSA-j3f7-7rmc-6wqj']
+    >>> get_aliases({"aliases": ["CVE-2021-40831"]})
+    ['CVE-2021-40831']
+    >>> get_aliases({"aliases": ["CVE-2022-22817", "GHSA-8vj2-vxx3-667w"], "id": "GHSA-j3f7-7rmc-6wqj"})
+    ['CVE-2022-22817', 'GHSA-8vj2-vxx3-667w', 'GHSA-j3f7-7rmc-6wqj']
+    """
+    vulnerability_id = raw_data.get("id")
+    vulnerability_aliases = raw_data.get("aliases") or []
+    if vulnerability_id:
+        vulnerability_aliases.append(vulnerability_id)
+    return vulnerability_aliases
+
+
+def get_published_date(raw_data):
+    published = raw_data.get("published")
+    return published and dateparser.parse(published)
+
+
+def get_severities(raw_data) -> []:
+    if "severity" in raw_data:
+        for sever_list in raw_data.get("severity") or []:
+            if sever_list.get("type") == "CVSS_V3":
+                yield VulnerabilitySeverity(
+                    system=SCORING_SYSTEMS["cvssv3.1_vector"], value=sever_list["score"]
+                )
+            else:
+                logger.error(f"NotImplementedError severity type- {raw_data['id']!r}")
+
+    ecosys = raw_data.get("ecosystem_specific") or {}
+    sever = ecosys.get("severity")
+    if sever:
+        yield VulnerabilitySeverity(
+            system=SCORING_SYSTEMS["generic_textual"],
+            value=sever,
+        )
+
+    database_specific = raw_data.get("database_specific") or {}
+    sever = database_specific.get("severity")
+    if sever:
+        yield VulnerabilitySeverity(
+            system=SCORING_SYSTEMS["generic_textual"],
+            value=sever,
+        )
+
+
+def get_references(raw_data, severities) -> []:
+    references = raw_data.get("references") or []
+    return [Reference(url=ref["url"], severities=severities) for ref in references if ref]
+
+
+def get_affected_purl(affected_pkg, raw_id):
+    package = affected_pkg.get("package") or {}
+    purl = package.get("purl")
+    if purl:
+        try:
+            return PackageURL.from_string(purl)
+        except ValueError:
+            logger.error(f"PackageURL ValueError - {raw_id !r} - purl: {purl !r}")
+
+    ecosys = package.get("ecosystem")
```

Comment:
I think there is a need to translate the "ecosystem" to the purl type as the values used by OSV are not exactly that of purl types.

Reply:
yes , we translated the "ecosystem" if the osv purl not found but what is wrong with osv purl "pkg:pypi/tensorflow-gpu"
Can you explain it  more?
```
>>> PackageURL.from_string("pkg:pypi/tensorflow-gpu") == PackageURL(type="pypi", name="tensorflow-gpu")
True 
```